skycoin/skywire
# add makefile directives

Did you run `make format && make check`?
yes

##### Changes:	
- `run-source` makefile directive write config to stdout & read config from stdin

##### Adds:
- `run-vpnsrv` makefile directive
- `run-source-test` makefile directive
- `run-vpnsrv-test` makefile directive
- `run-source-dmsghttp` makefile directive
- `run-source-dmsghttp-test` makefile directive
- `run-vpnsrv-dmsghttp` makefile directive
- `run-vpnsrv -dmsghttp-test` makefile directive


How to test this PR:

This PR comprises an efficient testing regime for the different possible configurations.
* with or without dmsghttp
* with or without test envs
* with or without vpn server enabled

The configs generated / used here have hypervisor enabled and the visor will attempt to launch the browser.

Also note that the permissions are correct in running it this way as user. The permissions will be elevated in the script that launches the apps, and the sudo password is cached initially to facilitate this.

default:
```
make run-source
```
default / vpnserver:
```
make run-vpnsrv
```
test
```
make run-source-test
```
test / vpn server
```
make run-vpsrv-test
```
dmsghttp
```
make run-source-dmsghttp
```
dmsghttp test
```
make run-source-dmsghttp-test
```
dmsghttp vpn server
```
make run-source-dmsghttp
```
dmsghttp vpn server test
```
make run-vpnsrv-dmsghttp-test
```
